torch/distributed/_functional_collectives.py

```diff
@@ -432,6 +432,9 @@ def reduce_scatter_tensor_coalesced(
 # Today, this maps 1:1 with "aten ops that are views".
 def _is_view_op(tgt):
     assert isinstance(tgt, torch._ops.OpOverload)
+    # Special case for `aten.to`. See issue: https://github.com/pytorch/pytorch/issues/133421
+    if "to" in tgt.__name__.split('.'):
+        return False
```

Comment:
hmm this feels like a bit of bandaid fix (although I agree it solves the `aten.to` problem).

The issue is that we have several composite operations that are "maybe-aliasing", and therefore can lie about their schemas. `aten.to.device` is an example: it may-or-may-not alias input, depending on whether the device matches the input tensor, but its schema reports as always aliasing.

Ordinarily, these ops will always decompose before we get to __torch_dispatch__. But under inference_mode, these ops can show up directly in __torch_dispatch__.

The way we've generally dealt with this for other subclasses is to force them to decompose these composite ops, by adding this code:

```
r = func.decompose(*args, **kwargs)
# this will attempt to run the eager-mode decomposition if one exists, and return NotImplemented otherwise
if r is not NotImplemented:
    return r
```

Although cc @weifengpy, I'm curious what you think, since this could have an impact on eager performance (we'd probably have to measure it)

Reply:
however, given that AsyncCollectiveTensor doesn't really do very much other than branch on view ops... another strategy that is more general than this PR but might be less risky for perf would just be to how `AsyncCollectiveTensor` assume that all of these composite ops are not views (worst case, it does an early sync when it doesn't have to, but most view ops are not composite anyway).

You can do it like this:
```
# don't apply the view optimization to any `CompositeImplicitAutograd` ops
if torch._C._dispatch_has_kernel_for_dispatch_key(func.name(), DispatchKey.CompositeImplicitAutograd):
    return False
```

fyi @weifengpy @wanchaol

Reply:
sry for the late reply, I've fixed it in a more general way according to your suggestion. @bdhirsh